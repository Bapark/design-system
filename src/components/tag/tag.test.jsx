import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Tag from './tag.jsx';

describe('Tag', () => {
  it('renders a tag component', () => {
    const tree = renderer
      .create((
        <Tag title="Test Title" />
      )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('uses a title', () => {
    render(<Tag title="Test Title" />);
    expect(screen.getByText('Test Title').tagName).toEqual('SPAN');
  });
});
