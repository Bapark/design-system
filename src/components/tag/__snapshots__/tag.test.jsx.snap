// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders a tag component 1`] = `
<div
  className="tag"
  role="row"
>
  <span
    className="title"
    role="gridcell"
  >
    Test Title
  </span>
  <span
    className="icon"
    role="gridcell"
  >
    <svg
      className="icon-base size-small"
      role="img"
    >
      <use
        xlinkHref="#icon-clear-small.svg"
      />
    </svg>
  </span>
</div>
`;
